File: code/string-algorithms/pangram_checker/pangram_checker.py
# Part of Cosmos by OpenGenus Foundation

def pangram_checker(text):
    return {chr(i) for i in range(ord('a'), ord('z') + 1)}.issubset(text.lower())

print(pangram_checker('hello')) # False
print(pangram_checker('The quick brown fox jumps over the lazy dog')) # True

File: code/string-algorithms/pangram_checker/pangram_checker.php
<?php

// Part of Cosmos by OpenGenus Foundation

function pangram_checker($text)
{
    $text = strtolower($text);
    $alphabet = array_fill(0, 26, false);
    $count = 0;
    $length = strlen($text);
    for ($i = 0; $i < $length; $i++) {
        $ord = ord($text[$i]);
        if ($ord >= 97 && $ord <= 122 && !$alphabet[$ord - 97]) {
            $alphabet[$ord - 97] = true;
            $count++;
        }
    }
    return $count === 26;
}

echo pangram_checker('hello') ? 'true' : 'false', "\n"; // false
echo pangram_checker('The quick brown fox jumps over the lazy dog') ? 'true' : 'false', "\n"; // true
